### PR TITLE
find external python and install ninja

### PR DESCRIPTION
**Pull request by loulawrence**
On windows, enable find external python and enable ninja package install.

### CODE REVIEW

File: var/spack/repos/builtin/packages/python/package.py [hunk 1]
```diff
@@ -642,6 +643,16 @@ def command(self):
             path = os.path.join(self.prefix.bin, 'python{0}'.format(ver))
             if os.path.exists(path):
                 return Executable(path)
+            else:
+                path = self.prefix.bin.replace(os.sep+"bin","")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/python/package.py]: I don't really understand this logic. To me it looks like this is equivalent to `self.prefix` (without the `.bin`). You don't even use this variable anyway, so you can delete it.

File: var/spack/repos/builtin/packages/python/package.py [hunk 2]
```diff
@@ -642,6 +643,16 @@ def command(self):
...
+                if sys.platform != "win32":
+                   path = os.path.join(self.prefix.bin, 'python{0}'.format(ver))
+                else:
+                   path = os.path.join(self.prefix, 'python{0}'.format(ver))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/python/package.py]:
```suggestion
                   path = os.path.join(self.prefix, 'python{0}.exe'.format(ver))
```
Then the next line isn't needed

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/python/package.py (hunk 1 above)
Comment (loulawrence) [reply var/spack/repos/builtin/packages/python/package.py]: Thanks for the review. This is left over from an earlier draft and should not be there.

File: var/spack/repos/builtin/packages/python/package.py [hunk 3]
```diff
@@ -642,6 +643,14 @@ def command(self):
             path = os.path.join(self.prefix.bin, 'python{0}'.format(ver))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/python/package.py]: It seems like your new logic should be combined with this existing logic.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**